drv/gimlet-seq-server/src/main.rs

```diff
@@ -1119,7 +1138,20 @@ fn read_spd_data_and_load_packrat(
             // We'll store that byte and then read 255 more.
             tmp[0] = first;
 
-            retry_i2c_txn(|| spd.read_into(&mut tmp[1..]))?;
+            let mut retried = false;
+
+            retry_i2c_txn(I2cTxn::SpdLoad(nbank, i), || {
+                if retried {
+                    //
+                    // If our read needs to be retried, we need to also reset
+                    // ourselves back to the 0th byte.
+                    //
+                    _ = spd.read_reg::<u8, u8>(0)?;
+                }
+
+                retried = true;
+                spd.read_into(&mut tmp[1..])
+            })?;
```

Comment:
It's cleaner to unconditionally read the entire register if the byte-0 read succeeds:
```rust
            // Try reading the first byte; if this fails, we will assume
            // the device isn't present.
            if spd.read_reg::<u8, u8>(0).is_ok() {
                present[usize::from(ndx)] = true;
                npresent += 1;

                // Now do the full read (re-reading byte 0 for simplicity)
                retry_i2c_txn(I2cTxn::SpdLoad(nbank, i), || {
                    spd.read_reg_into(0, &mut tmp)
                })?;

                packrat.set_spd_eeprom(ndx, false, 0, &tmp);
            } else {
                ringbuf_entry!(Trace::SpdAbsent(nbank, i, ndx));
            }
```
(and I don't think it's worth the extra complexity to avoid re-reading byte 0)

Reply:
There's a limit of 255 bytes on an i2c read, so this needs to be over two reads anyway, FWIW.